src/Opdex.Platform.WebApi/Controllers/IndexController.cs

```diff
@@ -99,21 +97,28 @@ public async Task<IActionResult> ResyncFromDeployment(ResyncFromDeploymentReques
 
     /// <summary>Rewind to Block</summary>
     /// <param name="request">Request to rewind back to specific block.</param>
+    /// <param name="cancellationToken">Cancellation token.</param>
     /// <response code="204">Indexer rewound.</response>
     [HttpPost("rewind")]
     [Authorize(Policy = "AdminOnly")]
     [ProducesResponseType(StatusCodes.Status204NoContent)]
     [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status401Unauthorized)]
     [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status403Forbidden)]
-    public async Task<ActionResult> Rewind(RewindRequest request)
+    public async Task<ActionResult> Rewind(RewindRequest request, CancellationToken cancellationToken)
     {
-        var tryLock = await _mediator.Send(new MakeIndexerLockCommand(IndexLockReason.Rewind), CancellationToken.None);
+        var latestSyncedBlock = await _mediator.Send(new RetrieveLatestBlockQuery(), cancellationToken);
+        if (latestSyncedBlock.Height <= request.Block) throw new InvalidDataException("block", "Block height must be less than current synced height.");
+
+        var tryLock = await _mediator.Send(new MakeIndexerLockCommand(IndexLockReason.Rewinding), CancellationToken.None);
         if (!tryLock) throw new IndexingAlreadyRunningException();
 
         try
         {
+            var blockHash = await _mediator.Send(new RetrieveCirrusBlockHashByHeightQuery(request.Block), cancellationToken);
+            var rewindBlock = await _mediator.Send(new RetrieveCirrusBlockReceiptByHashQuery(blockHash), cancellationToken);
             var rewound = await _mediator.Send(new CreateRewindToBlockCommand(request.Block), CancellationToken.None);
-            if (!rewound) throw new Exception("Indexer rewind unexpectedly failed.");
+            if (!rewound) throw new Exception($"Failure rewinding database to block height: {request.Block}");
+            await _mediator.Send(new ProcessLatestBlocksCommand(rewindBlock, _network), CancellationToken.None);
```

Comment:
- `blockHash` can be retrieved by height from the database using `request.Block` probably faster than asking the FN 
- Retrieving `blockHash` and `rewindBlock` can be moved below `if (!rewound) ...` since they aren't used yet and may not need to be if rewinding fails

Reply:
Took care of the first two, still thinking on the third.

Reply:
Thanks for taking care of it. I see the `ProcessLatestBlocks` is still there, did you decide your 3rd point wasn't an issue? I don't see that there's a timeout there